okhttp/src/main/java/com/squareup/okhttp/ConnectionSpec.java

```diff
@@ -219,7 +219,7 @@ private ConnectionSpec supportedSpec(SSLSocket sslSocket) {
     private String[] tlsVersions;
     private boolean supportsTlsExtensions;
 
-    Builder(boolean tls) {
+    public Builder(boolean tls) {
```

Comment:
Right, I'm just wondering if one wanted to use, say, CHACHA20, that's currently unavailable. The string API enables this. My proposal is that we'd encourage users to use the standard APIs rather than making decisions themselves, but still allow other users to make the selections they need if  they (think they) know that they're doing.